app/services/mapping/goodsshipment/PreviousDocumentsBuilder.scala

```diff
@@ -33,28 +33,47 @@ object PreviousDocumentsBuilder {
   def build(implicit cacheMap: CacheMap): util.List[GoodsShipment.PreviousDocument] =
     cacheMap
       .getEntry[PreviousDocumentsData](Document.formId)
+      .filter(isDefined)
       .map(_.documents.map(createPreviousDocuments))
       .getOrElse(Seq.empty)
       .toList
       .asJava
 
+  private def isDefined(previousDocumentsData: PreviousDocumentsData): Boolean =
+    previousDocumentsData.documents.nonEmpty && previousDocumentsData.documents.forall(
+      doc =>
+        !doc.goodsItemIdentifier.getOrElse("").isEmpty ||
```

Comment:
Instead of using `!` and `isEmpty` probably it will be easier to use nonEmpty

Comment for lines 66 and 71 too.

Reply:
:wink: Unfortunately, the `String` class does not have a `isNotEmpty` method and we are not using `Ruby` where we could override the `String` class.

Reply:
`nonEmpty`

Reply:
and as you so eloquently, demonstrated there is a `nonEmpty` method :smile: 
Awesome! 
Thank you so much! :)